### PR TITLE
Quick pass on some parameter naming

### PR DESCRIPTION
**Pull request by glemaitre**
Just a quick pass on some parameter naming.

### CODE REVIEW

File: sklearn/inspection/_h_statistic.py [hunk 1]
```diff
@@ -260,17 +268,16 @@ def h_statistic(
...
-    if sample_weight is not None:
-        sample_weight = _check_sample_weight(sample_weight, X)
+    sample_weight = _check_sample_weight(sample_weight, X)
 
     # Usually, the data is too large and we need subsampling
-    if X.shape[0] > n_max:
-        row_indices = sample_without_replacement(
-            n_population=X.shape[0], n_samples=n_max, random_state=random_state
+    if _num_samples(X) > subsample:
+        X, sample_weight = resample(
```
Comment (glemaitre) [inline sklearn/inspection/_h_statistic.py:277]: We are using this utility in some other places. This should avoid a bit of boilerplate code.

File: sklearn/inspection/_h_statistic.py [hunk 2]
```diff
@@ -169,16 +175,18 @@ def h_statistic(
...
-        See :term:`Glossary <random_state>`.
+    subsample : int, default=500
+        Maximum number of samples drawn without replacement for computational
+        efficiency. Reducing the number of samples improves speed but increases
+        variance of the estimate.
 
     eps : float, default=1e-10
         Threshold below which numerator values are set to 0.
 
+    random_state : int, RandomState instance, default=None
```
Comment (glemaitre) [inline sklearn/inspection/_h_statistic.py:188]: Usually we place the random state at the end

File: sklearn/inspection/_h_statistic.py [hunk 3]
```diff
@@ -280,16 +287,16 @@ def h_statistic(
         ).ravel()
 
     # CALCULATIONS
-    pd_univariate = []
-    for idx in feature_indices:
-        xxx = _calculate_pd_over_data(
+    pd_univariate = [
```
Comment (glemaitre) [inline sklearn/inspection/_h_statistic.py:292]: Let's write a list comprehension

**Comment by glemaitre** [comment]
@mayer79 Could you check if the changes make sense to you?

File: sklearn/inspection/_h_statistic.py (hunk 1 above)
Comment (mayer79) [reply sklearn/inspection/_h_statistic.py:277]: Looks good! I think I have picked some snippets from permutation importance.

**Review by mayer79 — APPROVED** [review]
Lgtm,  thx a lot

**Comment by glemaitre** [comment]
Cool. I'll try to push forward the on the computation. It might be tight for 1.6 but now that I have the understanding I'll make sure that we merge it for 1.7 for sure.

**Comment by mayer79** [comment]
No hurry 👍

**Pull request merged**